dGame/dComponents/ActivityComponent.cpp

```diff
@@ -30,23 +30,11 @@
 #include "LeaderboardManager.h"
 
 ActivityComponent::ActivityComponent(Entity* parent, int32_t activityID) : Component(parent) {
-	if (activityID > 0) m_ActivityID = activityID;
-	else m_ActivityID = parent->GetVar<int32_t>(u"activityID");
-	CDActivitiesTable* activitiesTable = CDClientManager::Instance().GetTable<CDActivitiesTable>();
-	std::vector<CDActivities> activities = activitiesTable->Query([this](CDActivities entry) {return (entry.ActivityID == m_ActivityID); });
-
-	for (CDActivities activity : activities) {
-		m_ActivityInfo = activity;
-		if (static_cast<Leaderboard::Type>(activity.leaderboardType) == Leaderboard::Type::Racing && Game::config->GetValue("solo_racing") == "1") {
-			m_ActivityInfo.minTeamSize = 1;
-			m_ActivityInfo.minTeams = 1;
-		}
-		if (m_ActivityInfo.instanceMapID == -1) {
-			const auto& transferOverride = parent->GetVarAsString(u"transferZoneID");
-			if (!transferOverride.empty()) {
-				GeneralUtils::TryParse(transferOverride, m_ActivityInfo.instanceMapID);
-			}
-		}
+	m_ActivityID = activityID;
+	LoadActivityData(activityID);
```

Comment:
Is there a reason not to do this only once after the `if`?

Reply:
This is precisely what the client does functionally
Use the component id as the default activity id and load its data from the database
if activityID is specified and if that column exists in the activities table, update the activity info with that data.
This is what actually resolves the NT parrot foot race issue and why DLU inserted a activity 49 into the database in the first place.  It used all of the data from 39 by default, tried to load 49, found nothing, and used 49 for the leaderboards and all the data for 39.  Will add a comment.